// @aws-amplify/ui-react-storage/dist/esm/components/StorageBrowser/useAction/utils.mjs
import { isFunction } from '@aws-amplify/ui';
import '@aws-amplify/storage/internals';
import 'aws-amplify/storage';
import '../actions/configs/context.mjs';
import { isDefaultActionViewType } from '../actions/configs/defaults.mjs';

const resolveHandler = (value) => (isFunction(value) ? value : value.handler);
const getActionHandlers = (configs) => {
    const { copy: copyConfig, createFolder: createFolderConfig, delete: deleteConfig, download: downloadConfig, upload: uploadConfig, listLocationItems, listLocations, } = configs.default;
    const defaultHandlers = {
        copy: copyConfig.handler,
        createFolder: createFolderConfig.handler,
        delete: deleteConfig.handler,
        download: 'handler' in downloadConfig ? downloadConfig.handler : downloadConfig,
        listLocationItems,
        listLocations,
        upload: uploadConfig.handler,
    };
    return !configs.custom
        ? defaultHandlers
        : Object.entries(configs.custom).reduce((handlers, [key, config]) => isDefaultActionViewType(key)
            ? handlers
            : { ...handlers, [key]: resolveHandler(config) }, defaultHandlers);
};

export { getActionHandlers };
