// @aws-amplify/ui-react-storage/dist/esm/components/StorageBrowser/views/context/actionViews.mjs
import React__default from 'react';
import { UploadView } from '../LocationActionView/UploadView/UploadView.mjs';
import '../../fileItems/context.mjs';
import '@aws-amplify/ui';
import '../../store/context.mjs';
import '../../useAction/context.mjs';
import '@aws-amplify/ui-react-core/elements';
import '../../credentials/context.mjs';
import '@aws-amplify/storage/internals';
import '../../configuration/context.mjs';
import '@aws-amplify/ui-react-core';
import 'aws-amplify/storage';
import '../../actions/configs/context.mjs';
import '../../actions/configs/defaults.mjs';
import { CreateFolderView } from '../LocationActionView/CreateFolderView/CreateFolderView.mjs';
import { CopyView } from '../LocationActionView/CopyView/CopyView.mjs';
import '../../locationItems/context.mjs';
import { DeleteView } from '../LocationActionView/DeleteView/DeleteView.mjs';
import { DownloadView } from '../LocationActionView/DownloadView/DownloadView.mjs';

const DEFAULT_ACTION_VIEWS = {
    createFolder: CreateFolderView,
    copy: CopyView,
    delete: DeleteView,
    download: DownloadView,
    upload: UploadView,
};
const ActionViewsContext = React__default.createContext({
    action: DEFAULT_ACTION_VIEWS,
});
function useActionViews() {
    return React__default.useContext(ActionViewsContext);
}

export { ActionViewsContext, DEFAULT_ACTION_VIEWS, useActionViews };
