// @aws-amplify/ui-react-storage/dist/esm/components/StorageBrowser/views/context/primaryViews.mjs
import React__default from 'react';
import '../LocationActionView/CopyView/CopyView.mjs';
import '@aws-amplify/ui';
import '../../locationItems/context.mjs';
import '../../store/context.mjs';
import '../../useAction/context.mjs';
import '@aws-amplify/ui-react-core/elements';
import '../../credentials/context.mjs';
import '@aws-amplify/storage/internals';
import '../../configuration/context.mjs';
import '@aws-amplify/ui-react-core';
import 'aws-amplify/storage';
import '../../actions/configs/context.mjs';
import '../../actions/configs/defaults.mjs';
import '../LocationActionView/CreateFolderView/CreateFolderView.mjs';
import '../LocationActionView/DeleteView/DeleteView.mjs';
import '../LocationActionView/DownloadView/DownloadView.mjs';
import { LocationActionView } from '../LocationActionView/LocationActionView.mjs';
import '../LocationActionView/UploadView/UploadView.mjs';
import '../../fileItems/context.mjs';
import { LocationDetailView } from '../LocationDetailView/LocationDetailView.mjs';
import { LocationsView } from '../LocationsView/LocationsView.mjs';

const DEFAULT_PRIMARY_VIEWS = {
    LocationActionView: LocationActionView,
    LocationDetailView: LocationDetailView,
    LocationsView: LocationsView,
};
const PrimaryViewsContext = React__default.createContext({
    primary: DEFAULT_PRIMARY_VIEWS,
});

export { DEFAULT_PRIMARY_VIEWS, PrimaryViewsContext };
