// @aws-amplify/ui-react-storage/dist/esm/components/StorageBrowser/actions/configs/defaults.mjs
import '@aws-amplify/storage/internals';
import '@aws-amplify/ui';
import { defaultHandlers } from '../handlers/defaults.mjs';
import 'aws-amplify/storage';

const copyActionConfig = {
    viewName: 'CopyView',
    actionListItem: {
        disable: (selected) => !selected || selected.length === 0,
        hide: (permissions) => !permissions.includes('write'),
        icon: 'copy-file',
        label: 'Copy',
    },
    handler: defaultHandlers.copy,
};
const deleteActionConfig = {
    viewName: 'DeleteView',
    actionListItem: {
        disable: (selected) => !selected || selected.length === 0,
        hide: (permissions) => !permissions.includes('delete'),
        icon: 'delete-file',
        label: 'Delete',
    },
    handler: defaultHandlers.delete,
};
const createFolderActionConfig = {
    viewName: 'CreateFolderView',
    actionListItem: {
        hide: (permissions) => !permissions.includes('write'),
        icon: 'create-folder',
        label: 'Create folder',
    },
    handler: defaultHandlers.createFolder,
};
const uploadActionConfig = {
    viewName: 'UploadView',
    actionListItem: {
        hide: (permissions) => !permissions.includes('write'),
        icon: 'upload-file',
        label: 'Upload',
    },
    handler: defaultHandlers.upload,
};
const downloadActionConfig = {
    viewName: 'DownloadView',
    actionListItem: {
        disable: (selected) => !selected || selected.length === 0,
        hide: (permissions) => !permissions.includes('get'),
        icon: 'download',
        label: 'Download',
    },
    handler: defaultHandlers.download,
};
// Action view configs only, does not include `listLocationItems`
const defaultActionViewConfigs = {
    copy: copyActionConfig,
    createFolder: createFolderActionConfig,
    download: downloadActionConfig,
    delete: deleteActionConfig,
    upload: uploadActionConfig,
};
const DEFAULT_ACTION_VIEW_TYPES = Object.keys(defaultActionViewConfigs);
const isDefaultActionViewType = (value) => DEFAULT_ACTION_VIEW_TYPES.some((type) => type === value);
const defaultActionConfigs = {
    ...defaultActionViewConfigs,
    listLocationItems: defaultHandlers.listLocationItems,
};

export { DEFAULT_ACTION_VIEW_TYPES, copyActionConfig, createFolderActionConfig, defaultActionConfigs, defaultActionViewConfigs, deleteActionConfig, downloadActionConfig, isDefaultActionViewType, uploadActionConfig };
