// @aws-amplify/ui-react-storage/dist/esm/components/StorageBrowser/views/LocationActionView/UploadView/useUploadView.mjs
import React__default from 'react';
import { useFileItems } from '../../../fileItems/context.mjs';
import '@aws-amplify/ui';
import { useStore } from '../../../store/context.mjs';
import '../../../useAction/context.mjs';
import { useAction } from '../../../useAction/useAction.mjs';
import '@aws-amplify/ui-react-core';
import '@aws-amplify/ui-react-core/elements';
import '../../../credentials/context.mjs';
import '@aws-amplify/storage/internals';
import '../../../configuration/context.mjs';
import 'aws-amplify/storage';
import '../../../actions/configs/context.mjs';
import '../../../actions/configs/defaults.mjs';
import { DEFAULT_OVERWRITE_ENABLED } from './constants.mjs';
import { usePaginate } from '../../hooks/usePaginate.mjs';

const DEFAULT_PAGE_SIZE = 100;
const useUploadView = (options) => {
    const { onExit: _onExit } = options ?? {};
    const [{ location }, storeDispatch] = useStore();
    const [{ validItems, invalidItems: invalidFiles }, fileItemsDispatch] = useFileItems();
    const { current } = location;
    const [isOverwritingEnabled, setIsOverwritingEnabled] = React__default.useState(DEFAULT_OVERWRITE_ENABLED);
    const items = React__default.useMemo(() => (validItems ?? []).map((item) => ({
        ...item,
        key: `${location.key}${item.key}`,
        preventOverwrite: !isOverwritingEnabled,
    })), [validItems, isOverwritingEnabled, location.key]);
    const [{ isProcessing, isProcessingComplete, statusCounts, tasks }, handleUploads,] = useAction('upload', { items });
    const { currentPage, handlePaginate, pageItems: pageTasks, } = usePaginate({
        items: tasks,
    });
    const onDropFiles = (files) => {
        if (files) {
            fileItemsDispatch({ type: 'ADD_FILES', files });
        }
    };
    const onSelectFiles = (type) => {
        fileItemsDispatch({ type: 'SELECT_FILES', selectionType: type });
    };
    const onActionStart = () => {
        handleUploads();
    };
    const onActionCancel = () => {
        tasks.forEach((task) => task.cancel?.());
    };
    const onActionExit = () => {
        // clear files state
        fileItemsDispatch({ type: 'RESET_FILE_ITEMS' });
        // clear selected action
        storeDispatch({ type: 'RESET_ACTION_TYPE' });
        _onExit?.(current);
    };
    const onToggleOverwrite = () => {
        setIsOverwritingEnabled((prev) => !prev);
    };
    const onTaskRemove = ({ data }) => {
        fileItemsDispatch({ type: 'REMOVE_FILE_ITEM', id: data.id });
    };
    return {
        isProcessing,
        isProcessingComplete,
        isOverwritingEnabled,
        location,
        invalidFiles,
        statusCounts,
        tasks: pageTasks,
        page: currentPage,
        hasNextPage: currentPage * DEFAULT_PAGE_SIZE < items.length,
        highestPageVisited: Math.ceil(items.length / DEFAULT_PAGE_SIZE),
        onActionCancel,
        onActionExit,
        onActionStart,
        onDropFiles,
        onPaginate: handlePaginate,
        onTaskRemove,
        onSelectFiles,
        onToggleOverwrite,
    };
};

export { useUploadView };
