// @aws-amplify/ui-react-storage/dist/esm/browser.mjs
import '@aws-amplify/storage/internals';
import '@aws-amplify/ui';
export { defaultHandlers } from './components/StorageBrowser/actions/handlers/defaults.mjs';
import 'aws-amplify/storage';
import './components/StorageBrowser/actions/configs/context.mjs';
export { defaultActionConfigs } from './components/StorageBrowser/actions/configs/defaults.mjs';
export { createAmplifyAuthAdapter } from './components/StorageBrowser/adapters/createAmplifyAuthAdapter/createAmplifyAuthAdapter.mjs';
export { createManagedAuthAdapter } from './components/StorageBrowser/adapters/createManagedAuthAdapter/createManagedAuthAdapter.mjs';
import 'react';
import '@aws-amplify/ui-react';
import './components/StorageBrowser/components/elements/definitions.mjs';
import './components/StorageBrowser/components/elements/IconElement.mjs';
import './components/StorageBrowser/components/composables/context.mjs';
import '@aws-amplify/ui-react-core';
import '@aws-amplify/ui-react-core/elements';
export { componentsDefault } from './components/StorageBrowser/components/defaults.mjs';
export { default as createStorageBrowser } from './components/StorageBrowser/createStorageBrowser/createStorageBrowser.mjs';
import './components/StorageBrowser/displayText/context.mjs';
export { DEFAULT_STORAGE_BROWSER_DISPLAY_TEXT as DefaultStorageBrowserDisplayText } from './components/StorageBrowser/displayText/libraries/en/default.mjs';
