// @aws-amplify/ui-react-storage/dist/esm/components/StorageBrowser/displayText/libraries/en/default.mjs
import { DEFAULT_CREATE_FOLDER_VIEW_DISPLAY_TEXT } from './createFolderView.mjs';
import { DEFAULT_COPY_VIEW_DISPLAY_TEXT } from './copyView.mjs';
import { DEFAULT_DELETE_VIEW_DISPLAY_TEXT } from './deleteView.mjs';
import { DEFAULT_LOCATION_DETAIL_VIEW_DISPLAY_TEXT } from './locationDetailView.mjs';
import { DEFAULT_LOCATIONS_VIEW_DISPLAY_TEXT } from './locationsView.mjs';
import { DEFAULT_UPLOAD_VIEW_DISPLAY_TEXT } from './uploadView.mjs';
import { DEFAULT_DOWNLOAD_VIEW_DISPLAY_TEXT } from './downloadView.mjs';

const DEFAULT_STORAGE_BROWSER_DISPLAY_TEXT = {
    CopyView: DEFAULT_COPY_VIEW_DISPLAY_TEXT,
    CreateFolderView: DEFAULT_CREATE_FOLDER_VIEW_DISPLAY_TEXT,
    DeleteView: DEFAULT_DELETE_VIEW_DISPLAY_TEXT,
    DownloadView: DEFAULT_DOWNLOAD_VIEW_DISPLAY_TEXT,
    LocationDetailView: DEFAULT_LOCATION_DETAIL_VIEW_DISPLAY_TEXT,
    LocationsView: DEFAULT_LOCATIONS_VIEW_DISPLAY_TEXT,
    UploadView: DEFAULT_UPLOAD_VIEW_DISPLAY_TEXT,
};

export { DEFAULT_STORAGE_BROWSER_DISPLAY_TEXT };
