// @aws-amplify/ui-react-storage/dist/esm/components/StorageBrowser/displayText/libraries/en/shared.mjs
const DEFAULT_ACTION_VIEW_DISPLAY_TEXT = {
    actionCancelLabel: 'Cancel',
    actionExitLabel: 'Exit',
    actionDestinationLabel: 'Destination',
    statusDisplayCanceledLabel: 'Canceled',
    statusDisplayCompletedLabel: 'Completed',
    statusDisplayFailedLabel: 'Failed',
    statusDisplayInProgressLabel: 'In progress',
    statusDisplayTotalLabel: 'Total',
    statusDisplayQueuedLabel: 'Not started',
    // empty by default
    tableColumnCancelHeader: '',
    tableColumnStatusHeader: 'Status',
    tableColumnFolderHeader: 'Folder',
    tableColumnNameHeader: 'Name',
    tableColumnTypeHeader: 'Type',
    tableColumnSizeHeader: 'Size',
};
const DEFAULT_LIST_VIEW_DISPLAY_TEXT = {
    loadingIndicatorLabel: 'Loading',
    searchSubmitLabel: 'Submit',
    searchClearLabel: 'Clear search',
    getDateDisplayValue: (date) => new Intl.DateTimeFormat('en-US', {
        month: 'short',
        day: 'numeric',
        hour: 'numeric',
        year: 'numeric',
        minute: 'numeric',
        hourCycle: 'h12',
    }).format(date),
};

export { DEFAULT_ACTION_VIEW_DISPLAY_TEXT, DEFAULT_LIST_VIEW_DISPLAY_TEXT };
