// @aws-amplify/ui-react-storage/dist/esm/components/StorageBrowser/actions/handlers/copy.mjs
import { copy } from '@aws-amplify/storage/internals';
import { constructBucket } from './utils.mjs';

const copyHandler = (input) => {
    const { config, data } = input;
    const { accountId: expectedBucketOwner, credentials, customEndpoint, } = config;
    const { key, sourceKey, lastModified, eTag } = data;
    const bucket = constructBucket(config);
    const source = {
        bucket,
        expectedBucketOwner,
        // Per S3 requirement, copy source must the URI encoded.
        // This is NOT added to Amplify JS v6 because it will be a breaking
        // change to suddenly introduce URI encode to copy API source.
        //
        // see: https://docs.aws.amazon.com/AmazonS3/latest/API/API_CopyObject.html#API_CopyObject_RequestSyntax
        path: sourceKey.split('/').map(encodeURIComponent).join('/'),
        notModifiedSince: lastModified,
        eTag,
    };
    const destination = {
        bucket,
        expectedBucketOwner,
        path: key,
    };
    const result = copy({
        source,
        destination,
        options: { locationCredentialsProvider: credentials, customEndpoint },
    });
    return {
        result: result
            .then(({ path }) => ({
            status: 'COMPLETE',
            value: { key: path },
        }))
            .catch((error) => {
            const { message } = error;
            return { error, message, status: 'FAILED' };
        }),
    };
};

export { copyHandler };
