// @aws-amplify/ui-react-storage/dist/esm/components/StorageBrowser/actions/handlers/download.mjs
import { getUrl } from '@aws-amplify/storage/internals';
import { constructBucket } from './utils.mjs';

function downloadFromUrl(fileName, url) {
    const a = document.createElement('a');
    a.href = url;
    a.download = fileName;
    a.target = '_blank';
    document.body.appendChild(a);
    a.click();
    document.body.removeChild(a);
}
const downloadHandler = ({ config, data: { key }, }) => {
    const { accountId, credentials, customEndpoint } = config;
    const result = getUrl({
        path: key,
        options: {
            bucket: constructBucket(config),
            customEndpoint,
            locationCredentialsProvider: credentials,
            validateObjectExistence: true,
            contentDisposition: 'attachment',
            expectedBucketOwner: accountId,
        },
    })
        .then(({ url }) => {
        downloadFromUrl(key, url.toString());
        return { status: 'COMPLETE', value: { url } };
    })
        .catch((error) => {
        const { message } = error;
        return { error, message, status: 'FAILED' };
    });
    return { result };
};

export { downloadHandler };
