// @aws-amplify/ui-react-storage/dist/esm/components/StorageBrowser/views/utils/tableResolvers/utils.mjs
import { humanFileSize } from '@aws-amplify/ui';
import { STATUS_ICONS } from './constants.mjs';

const getFileType = (value, fallback = '') => value.lastIndexOf('.') !== -1
    ? value.slice(value.lastIndexOf('.') + 1)
    : fallback;
const getCellName = (value) => 
// `value.split` always returns an array with at least one entry
// ensuring `.pop()` will always return a string
value.split('/').pop();
const getUploadCellFolder = ({ data: { file: { webkitRelativePath }, }, }, fallback = '-') => webkitRelativePath
    ? webkitRelativePath.slice(0, webkitRelativePath.lastIndexOf('/') + 1)
    : fallback;
const isCopyActionTask = (task) => 'sourceKey' in task.data;
const getFileDataCellFolder = (task) => {
    const targetKey = isCopyActionTask(task)
        ? task.data.sourceKey
        : task.data.key;
    const { fileKey } = task.data;
    return targetKey.slice(0, -fileKey.length);
};
const getUploadCellProgress = ({ progress, status, }) => {
    // prefer `progress` if available, 1 if status is complete, default 0
    const value = progress ?? (status === 'COMPLETE' ? 1 : 0);
    const displayValue = `${Math.round(value * 100)}%`;
    return { displayValue, value };
};
const getFileSize = (value, fallback = '-') => (!value ? fallback : humanFileSize(value, true));
const getFileDataCancelCellContent = (data) => {
    const { item, props } = data;
    const { cancel, status } = item;
    const { isProcessing, onTaskRemove } = props;
    const isQueued = status === 'QUEUED';
    // a task is removable prior to processing start. Including `isQueued` ensures
    // that `isRemovable` is `false` on all tasks processing complete
    const isRemovable = isQueued && !isProcessing;
    // a task is cancelable while processing is true, and the task has a cancel handler
    const isCancelable = isProcessing && !!cancel;
    const ariaLabel = `${isRemovable ? 'Remove' : 'Cancel'} item: ${getCellName(item.data.fileKey)}`;
    const isDisabled = !isRemovable && !isCancelable;
    // resolve to `undefined` if not cancelable or removable
    const onClick = !isCancelable && !isRemovable
        ? undefined
        : () => {
            if (isRemovable) {
                onTaskRemove?.(item);
                // do not run cancel handler on remove
                return;
            }
            if (isCancelable)
                cancel();
        };
    return { ariaLabel, isDisabled, onClick, icon: 'cancel' };
};
/**
 * Generates a unique key for a table cell based on the key and item id
 */
const getFileDataCellKey = ({ key, item, }) => `${key}-${item.data.id}`;
const name = (data) => {
    const key = getFileDataCellKey(data);
    const { item } = data;
    const text = item.data.fileKey;
    const icon = STATUS_ICONS[item.status];
    return { key, type: 'text', content: { icon, text } };
};
const folder = (data) => {
    const key = getFileDataCellKey(data);
    const text = getFileDataCellFolder(data.item);
    return { key, type: 'text', content: { text } };
};
const type = (data) => {
    const key = getFileDataCellKey(data);
    const { fileKey } = data.item.data;
    const text = getFileType(fileKey);
    return { key, type: 'text', content: { text } };
};
const size = (data) => {
    const key = getFileDataCellKey(data);
    const { size: value } = data.item.data;
    const displayValue = getFileSize(value);
    return { key, type: 'number', content: { value, displayValue } };
};
const cancel = (data) => {
    const key = getFileDataCellKey(data);
    const content = getFileDataCancelCellContent(data);
    return { key, type: 'button', content };
};

export { cancel, folder, getCellName, getFileDataCancelCellContent, getFileDataCellFolder, getFileDataCellKey, getFileSize, getFileType, getUploadCellFolder, getUploadCellProgress, name, size, type };
