// @aws-amplify/ui-react-storage/dist/esm/components/StorageBrowser/components/elements/definitions.mjs
import { defineBaseElement } from '@aws-amplify/ui-react-core/elements';
import './IconElement.mjs';

const DescriptionListElement = defineBaseElement({
    type: 'dl',
    displayName: 'DescriptionList',
});
const DescriptionTermElement = defineBaseElement({
    type: 'dt',
    displayName: 'DescriptionTerm',
});
const DescriptionDetailsElement = defineBaseElement({
    type: 'dd',
    displayName: 'DescriptionDetails',
});
const LabelElement = defineBaseElement({
    type: 'label',
    displayName: 'Label',
});
const NavElement = defineBaseElement({
    type: 'nav',
    displayName: 'Nav',
});
defineBaseElement({
    type: 'p',
    displayName: 'Text',
});
const OrderedListElement = defineBaseElement({
    type: 'ol',
    displayName: 'OrderedList',
});
defineBaseElement({
    type: 'ul',
    displayName: 'UnorderedList',
});
const ListItemElement = defineBaseElement({
    type: 'li',
    displayName: 'ListItem',
});
const TableElement = defineBaseElement({
    type: 'table',
    displayName: 'Table',
});
const TableDataCellElement = defineBaseElement({
    type: 'td',
    displayName: 'TableDataCell',
});
const TableRowElement = defineBaseElement({
    type: 'tr',
    displayName: 'TableRow',
});
const TableHeaderElement = defineBaseElement({
    type: 'th',
    displayName: 'TableHeader',
});
const TableHeadElement = defineBaseElement({
    type: 'thead',
    displayName: 'TableHead',
});
const TableBodyElement = defineBaseElement({
    type: 'tbody',
    displayName: 'TableBody',
});
const HeadingElement = defineBaseElement({
    type: 'h2',
    displayName: 'Title',
});
defineBaseElement({
    type: 'svg',
    displayName: 'ProgressBar',
});
const InputElement = defineBaseElement({
    type: 'input',
    displayName: 'Input',
});
const ButtonElement = defineBaseElement({ type: 'button', displayName: 'Button' });
const ViewElement = defineBaseElement({
    type: 'div',
    displayName: 'View',
});
const SpanElement = defineBaseElement({
    type: 'span',
    displayName: 'Span',
});

export { ButtonElement, DescriptionDetailsElement, DescriptionListElement, DescriptionTermElement, HeadingElement, InputElement, LabelElement, ListItemElement, NavElement, OrderedListElement, SpanElement, TableBodyElement, TableDataCellElement, TableElement, TableHeadElement, TableHeaderElement, TableRowElement, ViewElement };
