// @aws-amplify/ui-react-storage/dist/esm/components/StorageBrowser/views/LocationActionView/CopyView/getDestinationPickerTableData.mjs
const DESTINATION_PICKER_COLUMNS = [
    { key: 'name', type: 'sort', content: { label: 'Folder name' } },
];
const getDestinationPickerTableData = ({ prefix, path, folders, onSelectFolder, }) => {
    const rows = !folders
        ? []
        : folders.map(({ id, key }) => {
            const folderSubPath = key.slice(`${prefix ?? ''}${path}`.length);
            const folderLocationPath = key.slice(prefix.length);
            const row = {
                key: id,
                content: [
                    {
                        key: `${DESTINATION_PICKER_COLUMNS[0].key}-${id}`,
                        type: 'button',
                        content: {
                            icon: 'folder',
                            ariaLabel: folderSubPath,
                            label: folderSubPath,
                            onClick: () => {
                                onSelectFolder?.(id, folderLocationPath);
                            },
                        },
                    },
                ],
            };
            return row;
        });
    return {
        headers: DESTINATION_PICKER_COLUMNS,
        rows,
    };
};

export { getDestinationPickerTableData };
